gprofiler/main.py

```diff
@@ -723,6 +744,45 @@ def init_pid_file(pid_file: str) -> None:
     Path(pid_file).write_text(str(os.getpid()))
 
 
+# Set limits and return path of the cgroup.
+def set_limits(cpu: float, memory: int) -> str:
+    cgroups = {}
+    logger.debug("Check if cgroup version is supported.")
+    try:
+        cgroups["cpu"] = CpuCgroup()
+        cgroups["memory"] = MemoryCgroup()
+    except UnsupportedCGroupV2:
+        logger.error("cgroup v2 is not supported by gProfiler, cpu and memory limits wouldn't be set.")
+        return
+
+    logger.debug("Prepare gProfiler cpu cgroup.")
+    try:
+        cgroups["cpu"].move_to_cgroup("gprofiler", os.getpid())
+    except AlreadyInCgroup:
+        logger.warning("gProfiler have already a cpu group.")
+
+    logger.debug("Set cpu limit in the cgroup.")
+    cgroups["cpu"].set_cpu_limit_cores(cpu)
```

Comment:
```suggestion
    try:
        cgroups["cpu"].move_to_cgroup("gprofiler", os.getpid())
    except AlreadyInCgroup:
        logger.warning("gProfiler already has a cpu group")
    else:
        cgroups["cpu"].set_cpu_limit_cores(cpu)
```

also removed the "set cpu limit" log, I think it's too verbose. You're already logging once that you are about to set resource limits - no need to do so again. Log only in the error / skip case - failed to set limits OR decided not to set limits due to already in cgroup.

Reply:
Please make the same change for the memory part below.